official/resnet/keras/keras_common.py

```diff
@@ -34,6 +34,14 @@
 TRAIN_TOP_1 = 'training_accuracy_top_1'
 
 
+class BatchTimestamp(object):
+  """A structure to store batch time stamp."""
+
+  def __init__(self, batch_index):
+    self.batch_index = batch_index
+    self.timestamp = time.time()
```

Comment:
Pass in the time so that the time is exactly equal to the start or end time and not off even by a nano second.